pkg/eval/builtin_fn_io.go

```diff
@@ -787,9 +789,10 @@ func fopen(name string) (vals.File, error) {
 //
 // @cf fopen
 
-func fclose(f vals.File) error {
-	return f.Close()
-}
+// Deprecated, written in file module instead.
```

Comment:
The deprecation should be part of the documentation. See, for example:

https://github.com/elves/elvish/blob/29034c20e9a30e9826c9f9aeac309b4d519db08e/pkg/eval/builtin_fn_str.go#L177-L183

You also need to add appropriate lines to the `checkDeprecatedBuiltin` function in *pkg/eval/compiler.go*.